src/prefect/server/schemas/filters.py

```diff
@@ -1487,12 +1487,22 @@ class ArtifactFilterKey(PrefectFilterBaseModel):
         example="my-artifact-%",
     )
 
+    exists_: Optional[bool] = Field(
+        default=None,
+        description=(
+            "If `true`, only include artifacts with a non-null key. If `false`, "
+            "only include artifacts with a null key."
+        ),
+    )
+
     def _get_filter_list(self, db: "PrefectDBInterface") -> List:
         filters = []
         if self.any_ is not None:
             filters.append(db.Artifact.key.in_(self.any_))
         if self.like_ is not None:
             filters.append(db.Artifact.key.ilike(f"%{self.like_}%"))
+        if self.exists_ is True:
```

Comment:
would we ever want to explicitly query for artifacts without a key set?

Reply:
I'm pretty sure no, but @znicholasbrown do you disagree? We also have a `type` filter to get results (which don't have a key).

Reply:
Both are now allowed 👍